### DISCUSSION TITLE
AutoComplete  如何在按键回车时触发搜索功能

### DISCUSSION THREAD

**Discussion by chenJJ-88** [ant-design/ant-design#50337] [Q&A] [answered]
由于特殊的样式要求，自己没有在AutoComplete 里面嵌入<Input.Search /> 组件，现在需要有一个回车事件触发搜索功能，请问怎么做呢

**Comment by crazyair** [comment]
onKeyDown 事件自己执行搜索

**Reply by chenJJ-88** [reply]
请教一下这个要怎么去绑定onKeyDown呢，ref拿到的 是 
<img width="224" alt="image" src="https://github.com/user-attachments/assets/8587af4a-2932-45d0-b0ba-b7fc5595e690">

**Reply by Wxh16144** [reply, marked as answer]
这样吧？
<img width="481" alt="image" src="https://github.com/user-attachments/assets/cb506e24-daf7-41c5-93eb-50d8c4ca0d41">

**Reply by Wxh16144** [reply]
ref 的方式也可以这样：

```tsx

import React from 'react'
import { AutoComplete } from 'antd'
import { useEventListener } from 'ahooks'

function App() {
  const ref = React.useRef<HTMLInputElement>(null)

  useEventListener(
    'keydown',
    (e) => {
      if (e.key === 'Enter' || e.keyCode === 13) {
        alert('useEventListener')
      }
    },
    {
      target: ref.current?.nativeElement,
    }
  );

  return (
    <AutoComplete
      style={{ width: 200 }}
      ref={ref}
      options={[{ value: 'text' }]}
    />
  )
}

export default App

```

**Reply by chenJJ-88** [reply]
> 这样吧？ <img alt="image" width="481" src="https://private-user-images.githubusercontent.com/32004925/358196103-cb506e24-daf7-41c5-93eb-50d8c4ca0d41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MTk1NjUsIm5iZiI6MTcyMzcxOTI2NSwicGF0aCI6Ii8zMjAwNDkyNS8zNTgxOTYxMDMtY2I1MDZlMjQtZGFmNy00MWM1LTkzZWItNTBkOGM0Y2EwZDQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEwNTQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5ZDUyM2I2MTUyMWU5OTY4ZTkzMjUzNTNmMDI3ODk5MmM0MGE3NWY3YTMzY2UyN2Y2NTcxY2NlMDA1MTI4ZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.By1B-LTAGEoeyx4XRDt1Fg66E1qEpYcdWD3vvdVtnFY">

已经好了谢谢你🙏
